src/main/java/calculator/Calculator.java

```diff
@@ -0,0 +1,21 @@
+package calculator;
+
+public abstract class Calculator<T> {
+    protected enum Type{
```

Comment:
```suggestion
    private enum Type{
```

계산기의 타입이라고하는 것이 Calculator 클래스에 존재하고 있고, 이를 외부에 노출 할 의도가 없다면 private 접근제한자를 사용하는 것이 좋을 것 같습니다.

또, 내부에 Type을 정의해두고 외부의 선택적인 메서드 호출로 Type이 정의된다면 외부에서 파라미터 값을 넘겨서 결정되는 방식으로 구현하는 것이 코드의 재사용성 측면에서 이점을 볼 수 있을 것 같습니다.

결국 정리하면 타입이라는 속성이 Calculator라는 클래스 내에서 관리되어야하는 속성인지가 고민됩니다.

어떻게 생각하시는지 의견 듣고 싶습니다.

Reply:
Calculator 클래스가 최종적으로 사용하게 되는 계산기 클래스이다보니 최대한 간단하게 객체를 생성해서 사용할 수 있는 방법을 고민하다가 생각해 낸 방식이었는데 말씀해주신대로 Type을 외부로 빼고 파라미터로 넘기는 것이 더 좋을 것 같습니다. Type은 다른데서 쓰는 곳이 따로 없어서 Calculator에 넣었는데 역시 굳이 Calculator 내부에서 변경될 필요가 없는 것 같습니다!

Reply:
Type enum을 CalculatorType이라는 enum으로 외부에 분리했는데 분리하고 보니 실제 계산 처리를 위한 Processor를 Calculator가 생성될 때 인자로 받아 계산시에 사용한다면 별도로 계산기의 타입을 관리할 필요가 없을 것 같아서 Calculator의 구조를 변경하고 Type을 없앴습니다.